src/scikit_build_core/cli/main.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import pathlib
+import sys
+
+import click
+
+from .. import __version__
+from .._compat.importlib import metadata
+
+
+@click.group("skbuild")
+@click.version_option(__version__)
+@click.option(
+    "--root",
+    "-r",
+    type=click.Path(
+        exists=True,
+        file_okay=False,
+        dir_okay=True,
+        writable=True,
+        path_type=pathlib.Path,
+    ),
+    help="Path to the python project's root",
+)
+@click.pass_context
+def skbuild(ctx: click.Context, root: pathlib.Path) -> None:  # noqa: ARG001
+    """
+    scikit-build Main CLI interface
+    """
+    # TODO: Add specific implementations
```

Comment:
I am thinking that here in `main` we load the project and make it available through `ctx` or other click decorators.

Any logger should also be initialized here. Should I just import `logger` from `_logging` and edit the levels and other settings in this section?

Reply:
Implementation:
https://github.com/scikit-build/scikit-build-core/blob/9fdd0a3f655215e280fb05b8749c5d5ce8712c55/src/scikit_build_core/settings/skbuild_read_settings.py#L99

Reply:
Logging is a global, you can just getLogger it.